control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -1055,7 +1088,20 @@ impl SpecOperationsHelper for VolumeSpec {
             }
             VolumeOperation::CreateSnapshot(_) => Ok(()),
             VolumeOperation::DestroySnapshot(_) => Ok(()),
-            VolumeOperation::Resize(_) => todo!(),
+            VolumeOperation::Resize(req_size) => {
+                if self.size > *req_size {
+                    return Err(SvcError::InvalidArguments {});
+                }
+                // Do Pre-checks for volume resize.
+                if !vol_status_ok_for_resize(self) {
+                    return Err(SvcError::InUse {
+                        kind: ResourceKind::Volume,
+                        id: self.uuid.to_string(),
+                    });
+                }
```

Comment:
Should we move the error to the function itself?
```suggestion
                vol_status_ok_for_resize(self)?;
```

Reply:
Let's keep it as is since we may have to modify function body for published volumes possibly later on, so we can do diverse error handling based on returned `bool`.

Reply:
The problem is that it doesn't add up, you're check if volume is ok for resize generically, but on the error case you return a specific "inUse", which means that actually the code is already assuming that ok for resize means in published/in use, see what I mean?

Reply:
hmm, ok. Modified. Will update diff on PR soon.